src/network/networkd-network.c

```diff
@@ -79,6 +79,47 @@ void network_config_section_free(NetworkConfigSection *cs) {
           free(cs);
 }
 
+/* Set defaults following RFC7844 */
+int network_apply_anonymize_if_set(Network *network) {
+        if (network->dhcp_anonymize) {
+                /* RFC7844 3.7
+                 SHOULD NOT send the Host Name option */
+                network->dhcp_send_hostname = false;
+                /* RFC7844 section 3.:
+                 MAY contain the Client Identifier option
+                 Section 3.5:
+                 clients MUST use client identifiers based solely
+                 on the link-layer address */
+                /* NOTE: Using MAC, as it does not reveal extra information,
+                * and some servers might not answer if this option is not sent */
+                network->dhcp_client_identifier = DHCP_CLIENT_ID_MAC;
+                /* RFC 7844 3.10:
+                 SHOULD NOT use the Vendor Class Identifier option */
+                /* NOTE: it was not initiallized to any value in network_load_one. */
+                network->dhcp_vendor_class_identifier = false;
+                /* RFC7844 section 3.6.:
+                 The client intending to protect its privacy SHOULD only request a
+                 minimal number of options in the PRL and SHOULD also randomly shuffle
+                 the ordering of option codes in the PRL.  If this random ordering
+                 cannot be implemented, the client MAY order the option codes in the
+                 PRL by option code number (lowest to highest).
+                */
+                /* NOTE: dhcp_use_mtu is false by default,
+                * though it was not initiallized to any value in network_load_one.
+                * Maybe there should be another var called *send*?
+                * (to use the MTU sent by the server but to do not send
+                * the option in the PRL). */
+                network->dhcp_use_mtu = false;
+                /* RFC7844 section 3.6.
+                * same comments as previous option */
+                network->dhcp_use_routes = false;
+                /* RFC7844 section 3.6.
+                * same comments as previous option */
+                network->dhcp_use_timezone = false;
+        }
+        return 0;
```

Comment:
given that this function can't fail, it should probably just be "void", not "int"